### PR TITLE
improve pyramid guessing

### LINKED ISSUE #699: pyramid checking seems to be breaking lazy loading
## 🐛 Bug
I was just trying to familiarize myself with napari's handling of zarr datasets using one of my big light sheet datasets... but it would hang indefinitely each time I added it.

I went back to the example in `zarr_nD_image.py`, but there too, as i increase the size of the zarr store, napari starts to hang upon `napari.view_image(data)`... regardless of whether I provide the `contrast_limits`, `rgb`  or  `is_pyramid=False` arguments.  Digging a bit more, it seems that, as of #556, it is impossible to avoid the callchain from `Image.__init__()` -> `get_pyramid_and_rgb()` -> `is_pyramid()` ... and that `is_pyramid()` call takes forever on a big zarr store.

Is this a regression or is `zarr_nD_image.py` now invoking `napari.view_image` incorrectly?

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR will close #699 and #710 and help with #733. See #736 for a meta issue describing our pyramid guessing and other big data problems right now.

It still won't solve all problems, for example those in #733 related to the auto-pyramid generation and `contrast_limits`, but it is a relatively minimal change for fixing a lot of bugs.

I've also added a bunch of tests, including 2 which currently fail but I skip because I think they will be important for us to ultimately support

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: napari/layers/image/tests/test_big_image_timing.py
Comment (jni) [inline napari/layers/image/tests/test_big_image_timing.py]: @sofroniewn the array here should actually be a zarr array and not dask: dask always worked fine with `is_pyramid()`. If you use `zarr.zeros` it should not take up any memory.

File: napari/util/tests/test_misc.py
Comment (jni) [inline napari/util/tests/test_misc.py]: Can you make these separate tests?

**Review by jni — APPROVED** [review]
@sofroniewn I'm approving because I'm about to go AFK, but I made what I think is a major comment: the tests should use zarr arrays, rather than dask arrays, because dask always worked: it's zarr arrays that were instantiated by `[d.shape for d in data]`.

**Comment by jni** [comment]
I recommend making sure that whatever tests you add don't pass on the current master before merging.

File: napari/layers/image/tests/test_big_image_timing.py
Comment (sofroniewn) [reply napari/layers/image/tests/test_big_image_timing.py]: hmm I do rememeber that, I had just checked that both those fail with a big dask array too. I'll add zarr.zeros too. Nice to check both

File: napari/util/tests/test_misc.py
Comment (sofroniewn) [reply napari/util/tests/test_misc.py]: will do

Comment (sofroniewn) [reply napari/util/tests/test_misc.py]: will use the timing functionality too

Comment (sofroniewn) [reply napari/util/tests/test_misc.py]: improved

**Comment by sofroniewn** [comment]
@jni this has now been improved and includes the changes from #692. The added tests previously failed, and now test zarr and dask. I will merge once the tests pass

**Comment by sofroniewn** [comment]
merging now

**Pull request merged**
